syslog/io_modules/decoders/syslog.lua

```diff
@@ -103,6 +107,11 @@ function decode(data, dh, mutable)
     fields.msg = nil
     sdu.add_fields(msg, fields)
 
+    if cfg.keep_data ~= nil then
```

Comment:
Move this to after line 104 and it can be simplified like the following
```lua
if cfg.keep_data then fields[cfg.keep_data] = data end
```

Reply:
Cheers. Will do